sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -115,7 +120,46 @@ public HttpLoggingPolicy(HttpLogDetailLevel detailLevel, boolean prettyPrintJSON
         return reqBodyLoggingMono;
     }
 
-    private Function<HttpResponse, Mono<HttpResponse>> logResponseDelegate(final ClientLogger logger, final URL url,
+    private void formatAllowableHeaders(Set<String> allowedHeaderNames, HttpRequest request, ClientLogger logger) {
+        StringBuilder sb = new StringBuilder();
+        for (HttpHeader header : request.getHeaders()) {
+            sb.append(header.getName()).append(":");
+            if (allowedHeaderNames.contains(header.getName())) {
+                sb.append(header.getValue());
+            } else {
+                sb.append(REDACTED_PLACEHOLDER);
+            }
+        }
+        logger.info(sb.toString());
```

Comment:
Should there be a newline or whitespace included in this concatenation operation?